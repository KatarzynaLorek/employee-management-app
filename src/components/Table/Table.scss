.table {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 400px;
}

@media (max-width: 767px) {
  .table {
    height: auto;
  }
}
