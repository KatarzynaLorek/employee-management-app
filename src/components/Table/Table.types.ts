import { IResponseObject } from '../../types/responses';

export interface ITable {
  handleUpdateButtonClick: (employee: IResponseObject) => void;
  handleDeleteButtonClick: (employee: IResponseObject) => void;
  data: IResponseObject[];
}
