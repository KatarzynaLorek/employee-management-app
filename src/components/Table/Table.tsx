import React from 'react';
import TableRow from './TableRow/TableRow';
import './Table.scss';
import { IResponseObject } from '../../types/responses';

interface ITableProps {
  handleOpenUpdateForm: (employee: IResponseObject | undefined) => void;
  tableData: IResponseObject[];
}

const Table = ({ handleOpenUpdateForm, tableData }: ITableProps): JSX.Element => (
  <div className="table">
    <TableRow handleClick={() => null}></TableRow>
    {tableData.map((item) => (
      <TableRow key={item.id} employeeData={item} handleClick={handleOpenUpdateForm}></TableRow>
    ))}
  </div>
);

export default Table;
