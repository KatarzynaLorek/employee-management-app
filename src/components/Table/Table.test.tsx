import React from 'react';
import { render } from '@testing-library/react';
import Table from './Table';

describe('Table component', () => {
  it('renders properly', () => {
    const { getAllByText } = render(<Table />);
    expect(getAllByText('Name')).toHaveLength(6);
  });
});
