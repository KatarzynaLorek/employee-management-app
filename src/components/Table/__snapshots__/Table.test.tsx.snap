// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders properly with 3 rows 1`] = `
<body>
  <div>
    <div
      class="table"
    >
      <div
        class="row--title"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            First Name
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            Last Name
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            Department
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            Position
          </p>
        </div>
        <div
          class="row__part"
        >
          Actions
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Table component renders properly with 5 rows 1`] = `
<body>
  <div>
    <div
      class="table"
    >
      <div
        class="row--title"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            First Name
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            Last Name
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            Department
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            Position
          </p>
        </div>
        <div
          class="row__part"
        >
          Actions
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            First Name
          </p>
          <p>
            mockFirstName
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Last Name
          </p>
          <p>
            mockLastname
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Department
          </p>
          <p>
            IT
          </p>
        </div>
        <div
          class="row__part"
        >
          <p
            class="row__label"
          >
            Position
          </p>
          <p>
            junior
          </p>
        </div>
        <div
          class="row__part"
        >
          <button
            class="row__button--update"
          >
            Update
          </button>
          <button
            class="row__button--delete"
          >
            Delete
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
